trino/trino_test.go

```diff
@@ -249,6 +249,82 @@ func TestQueryForUsername(t *testing.T) {
 	}
 }
 
+type TestQueryProgressCallback struct{}
+
+func (*TestQueryProgressCallback) ProgressUpdated(qr QueryProgressCallbackResponse) {
+	fmt.Printf("%+v, %+v \n", qr.QueryStats.State, qr.QueryStats.ProgressPercentage)
+}
+
+func TestQueryProgressWithCallback(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping test in short mode.")
+	}
+	c := &Config{
+		ServerURI:         *integrationServerFlag,
+		SessionProperties: map[string]string{"query_priority": "1"},
+	}
+
+	dsn, err := c.FormatDSN()
+	require.NoError(t, err)
+
+	db, err := sql.Open("trino", dsn)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		assert.NoError(t, db.Close())
+	})
+
+	callback := &TestQueryProgressCallback{}
+
+	rows, err := db.Query("SELECT 2", sql.Named("X-Query-Progress-Callback", callback))
+	require.NoError(t, err, "Failed executing query")
+	assert.NotNil(t, rows)
+
+	for rows.Next() {
+		var ts string
+		require.NoError(t, rows.Scan(&ts), "Failed scanning query result")
+
+		assert.Equal(t, "2", ts, "Expected value does not equal result value")
+	}
+}
+
+func TestQueryProgressWithCallbackPeriod(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping test in short mode.")
+	}
+	c := &Config{
+		ServerURI:         *integrationServerFlag,
+		SessionProperties: map[string]string{"query_priority": "1"},
+	}
+
+	dsn, err := c.FormatDSN()
+	require.NoError(t, err)
+
+	db, err := sql.Open("trino", dsn)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		assert.NoError(t, db.Close())
+	})
+
+	callback := &TestQueryProgressCallback{}
+	callbackPeriod, err := time.ParseDuration("100ms")
```

Comment:
Same here, you should actually assert that the callback does not get called too often/early. Maybe instead of just a counter, the `TestQueryProgressCallback` should track the minimum interval since the last update. The period should be much smaller so the test can finish quicker.

Reply:
how about this? check for queue and finished as it is always called.
```go
type TestQueryProgressCallback struct {
	statusMap map[string]float32
}

func (qpc *TestQueryProgressCallback) Update(qr QueryProgressCallbackResponse) {
	qpc.statusMap[qr.QueryStats.State] = qr.QueryStats.ProgressPercentage
}
...

assert.Contains(t, statusMap, "QUEUED")
assert.Contains(t, statusMap, "FINISHED")

Reply:
That doesn't assert if the period is used at all.

Reply:
If you have time, PTAL :)